src/PackageJsonSynchronizer.php

```diff
@@ -21,15 +21,37 @@
 class PackageJsonSynchronizer
 {
     private $rootDir;
+    private $vendorDirname;
 
-    public function __construct(?string $rootDir)
+    public function __construct(string $rootDir, string $vendorDirname)
     {
         $this->rootDir = $rootDir;
+        $this->vendorDirname = $vendorDirname;
     }
 
-    public function shouldSynchronize(): bool
+    public function shouldSynchronize(array $packagesNames): bool
     {
-        return $this->rootDir && file_exists($this->rootDir.'/package.json');
+        if (!$this->rootDir || !file_exists($this->rootDir.'/package.json')) {
+            return false;
+        }
+
+        foreach ($packagesNames as $packageName) {
+            if (!$this->resolveAssetsDir($packageName)) {
+                continue;
+            }
+
+            if (!file_exists($this->rootDir.'/'.$this->vendorDirname.'/'.$packageName.'/composer.json')) {
```

Comment:
this is compatible only with packages installed in the default location. Why not relying on Composer to get the installation path of the package instead ?

Reply:
`$this->vendorDirname` comes from Composer already

Reply:
But when using custom installers, not all packages are installed in `$vendorDir/$packageName` (btw, even with the default installer, Composer has the legacy `target-dir` feature which might change that too).

Reply:
I suggest we handle this case if someone has the issue.